editor/plugins/polygon_2d_editor_plugin.cpp

```diff
@@ -240,127 +236,76 @@ void Polygon2DEditor::_update_bone_list() {
 		cb->connect(SceneStringName(pressed), callable_mp(this, &Polygon2DEditor::_bone_paint_selected).bind(i));
 	}
 
-	uv_edit_draw->queue_redraw();
+	canvas->queue_redraw();
 }
 
 void Polygon2DEditor::_bone_paint_selected(int p_index) {
-	uv_edit_draw->queue_redraw();
+	canvas->queue_redraw();
 }
 
-void Polygon2DEditor::_uv_edit_mode_select(int p_mode) {
-	if (p_mode == 0) { //uv
-
-		uv_button[UV_MODE_CREATE]->hide();
-		uv_button[UV_MODE_CREATE_INTERNAL]->hide();
-		uv_button[UV_MODE_REMOVE_INTERNAL]->hide();
-		for (int i = UV_MODE_EDIT_POINT; i <= UV_MODE_SCALE; i++) {
-			uv_button[i]->show();
-		}
-		uv_button[UV_MODE_ADD_POLYGON]->hide();
-		uv_button[UV_MODE_REMOVE_POLYGON]->hide();
-		uv_button[UV_MODE_PAINT_WEIGHT]->hide();
-		uv_button[UV_MODE_CLEAR_WEIGHT]->hide();
-		_uv_mode(UV_MODE_EDIT_POINT);
-
-		bone_scroll_main_vb->hide();
-		bone_paint_strength->hide();
-		bone_paint_radius->hide();
-		bone_paint_radius_label->hide();
-	} else if (p_mode == 1) { //poly
-
-		for (int i = 0; i <= UV_MODE_SCALE; i++) {
-			uv_button[i]->show();
-		}
-		uv_button[UV_MODE_ADD_POLYGON]->hide();
-		uv_button[UV_MODE_REMOVE_POLYGON]->hide();
-		uv_button[UV_MODE_PAINT_WEIGHT]->hide();
-		uv_button[UV_MODE_CLEAR_WEIGHT]->hide();
-		if (node->get_polygon().is_empty()) {
-			_uv_mode(UV_MODE_CREATE);
-		} else {
-			_uv_mode(UV_MODE_EDIT_POINT);
-		}
-
-		bone_scroll_main_vb->hide();
-		bone_paint_strength->hide();
-		bone_paint_radius->hide();
-		bone_paint_radius_label->hide();
-	} else if (p_mode == 2) { //splits
-
-		for (int i = 0; i <= UV_MODE_SCALE; i++) {
-			uv_button[i]->hide();
-		}
-		uv_button[UV_MODE_ADD_POLYGON]->show();
-		uv_button[UV_MODE_REMOVE_POLYGON]->show();
-		uv_button[UV_MODE_PAINT_WEIGHT]->hide();
-		uv_button[UV_MODE_CLEAR_WEIGHT]->hide();
-		_uv_mode(UV_MODE_ADD_POLYGON);
-
-		bone_scroll_main_vb->hide();
-		bone_paint_strength->hide();
-		bone_paint_radius->hide();
-		bone_paint_radius_label->hide();
-	} else if (p_mode == 3) { //bones´
-
-		for (int i = 0; i <= UV_MODE_REMOVE_POLYGON; i++) {
-			uv_button[i]->hide();
-		}
-		uv_button[UV_MODE_PAINT_WEIGHT]->show();
-		uv_button[UV_MODE_CLEAR_WEIGHT]->show();
-		_uv_mode(UV_MODE_PAINT_WEIGHT);
-
-		bone_scroll_main_vb->show();
-		bone_paint_strength->show();
-		bone_paint_radius->show();
-		bone_paint_radius_label->show();
-		_update_bone_list();
-		bone_paint_pos = Vector2(-100000, -100000); //send brush away when switching
+void Polygon2DEditor::_select_mode(int p_mode) {
```

Comment:
Made this function more concise by hiding all toolbar elements and then reshowing them depending on mode. Old method was error prone, whenever new controls would be added to some mode there was the risk of forgetting to hide it in other modes. Now every mode simply enables the tools it needs.

Reply:
Hiding and reappearing causes additional redraws etc.
Not critical, but there might be a better way to track this, like making a HashSet of used buttons for mode and hiding all buttons that aren't in the set.